spec/services/pii/nist_encryption_spec.rb

```diff
@@ -0,0 +1,152 @@
+require 'rails_helper'
+# NOTE this set of specs intentionally includes lots of
+# duplicated code in order to explicitly show the algorithm at work.
+
+describe 'NIST Encryption Model' do
+# Generate and store a 128-bit salt S.
+# Z1, Z2 = scrypt(S, password)   # split 256-bit output into two halves
+# Generate random R.
+# D = KMS_GCM_Encrypt(key=server_secret, plaintext=R) ^ Z1
+# E = hash( Z2 + R )
+# F = hash(E)
+# C = GCM_Encrypt(key = E, plaintext=PII)  #occurs outside AWS-KMS
+# Store F in password file, and store C and D.
+#
+# To decrypt PII and to verify passwords:
+# Compute Z1’, Z2’ = scrypt(S, password’)
+# R’ = KMS_GCM_Decrypt(key=server_secret, ciphertext=(D ^ Z1*)).
+# E’ = hash( Z2’ + R’)
+# F’ = hash(E’)
+# Check to see if F’ matches the entry in the password file; if so, allow the login.
+# plaintext_PII = GCM_Decrypt(key=E’, ciphertext = C)
+
+  before do
+    allow(FeatureManagement).to receive(:use_kms?).and_return(true)
+  end
+
+  describe 'password hashing' do
+    it 'creates two substrings Z1 and Z2 via scrypt of password and salt' do
+      # Generate and store a 128-bit salt S.
+      ## (ours is actually 256 bits)
+      password = 'a long sekrit'
+      salt = Pii::Cipher.random_key
+
+      expect(salt.bytes.length).to eq 32
+
+      # Z1, Z2 = scrypt(S, password)   # split 256-bit output into two halves
+      user_access_key = UserAccessKeyMaker.hash(password, salt)
```

Comment:
@dogweather there is now a spec for `UserAccessKey` which includes showing the usage of `UserAccessKeyMaker` (which has only one method).